### PR TITLE
Feature/add bu info support

### PR DESCRIPTION
**Pull request by NickNYU**
Changes:
1. Add BU(Organization) Table in DB
2. Add Spring scheduled task, to retrieve organization info.
3. Add organization service loader to decoupling the implementation of retrieve organization infos
4. Add CAT monitor wrap the process of calling organization service loader
5. Update GUI(AngularJS) part, to add organization and cluster admin emails option during **creating&updating** Dc Cluster
6. Update GUI, to allow **organization & cluster admin email** shown in cluster list page
7. Update MetaUpdate for the new cluster creation logic

### CODE REVIEW

File: core/src/main/java/com/ctrip/xpipe/api/organization/Organization.java [hunk 1]
```diff
@@ -0,0 +1,15 @@
+package com.ctrip.xpipe.api.organization;
+
+import com.ctrip.xpipe.api.lifecycle.Ordered;
+import com.ctrip.xpipe.utils.ServicesUtil;
+
+import java.util.List;
+
+/**
+ * Created by zhuchen on 2017/8/30.
+ */
```
Comment (wenchaomeng) [inline core/src/main/java/com/ctrip/xpipe/api/organization/Organization.java]: comment codetemplates.xml

File: redis/redis-console/src/main/java/com/ctrip/xpipe/redis/console/config/ConsoleConfig.java [hunk 2]
```diff
@@ -48,5 +48,4 @@
 	Map<String, String> getConsoleCnameToDc();
 
 	QuorumConfig  getDefaultSentinelQuorumConfig();
-
```
Comment (wenchaomeng) [inline redis/redis-console/src/main/java/com/ctrip/xpipe/redis/console/config/ConsoleConfig.java]: dont commit

File: redis/redis-console/src/main/java/com/ctrip/xpipe/redis/console/config/impl/DefaultConsoleConfig.java [hunk 3]
```diff
@@ -38,6 +38,8 @@
 
     public static final String KEY_SENTINEL_QUORUM = "console.sentinel.quorum";
 
+
+
```
Comment (wenchaomeng) [inline redis/redis-console/src/main/java/com/ctrip/xpipe/redis/console/config/impl/DefaultConsoleConfig.java]: dont commit

File: redis/redis-console/src/main/java/com/ctrip/xpipe/redis/console/controller/api/data/MetaUpdate.java [hunk 4]
```diff
@@ -85,18 +88,48 @@ public RetMessage createCluster(@RequestBody ClusterCreateInfo outerClusterCreat
         }
 
         ClusterModel clusterModel = new ClusterModel();
-
+        OrganizationTbl organizationTbl;
+        try {
+            organizationTbl = getOrganizationId(clusterCreateInfo);
```
Comment (wenchaomeng) [inline redis/redis-console/src/main/java/com/ctrip/xpipe/redis/console/controller/api/data/MetaUpdate.java]: clusterCreateInfo.check();

File: redis/redis-console/src/main/java/com/ctrip/xpipe/redis/console/controller/api/data/MetaUpdate.java [hunk 5]
```diff
@@ -209,4 +245,15 @@ public RetMessage createShards(@PathVariable String clusterName, @RequestBody Li
         return result;
     }
 
+    private void checkNotNull(final Object o, String errorPrefix) {
+        if(o == null) {
+            throw new IllegalStateException(errorPrefix + " should not be null");
```
Comment (wenchaomeng) [inline redis/redis-console/src/main/java/com/ctrip/xpipe/redis/console/controller/api/data/MetaUpdate.java]: use common methods

File: redis/redis-console/src/main/java/com/ctrip/xpipe/redis/console/schedule/ScheduledOrganizationService.java [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+
+    @Autowired
+    OrganizationService organizationService;
+
+    private static final SimpleDateFormat dateFormat = new SimpleDateFormat("YYYY-MM-dd HH:mm:ss");
+
+
+    @Scheduled(fixedRate = XPipeConsoleConstant.SCHEDULED_ORGANIZATION_SERVICE)
+    public void updateOrganizations() {
+        logger.info("[updateOrganizations] update organization table @{}", dateFormat.format(new Date()));
```
Comment (wenchaomeng) [inline redis/redis-console/src/main/java/com/ctrip/xpipe/redis/console/schedule/ScheduledOrganizationService.java]: exception can continue?

File: redis/redis-console/src/test/java/com/ctrip/xpipe/redis/console/dao/ClusterDaoTest.java [hunk 7]
```diff
@@ -35,4 +43,32 @@ public void testCreateCluster() throws DalException {
...
+
+    @Test
+    public void testFindAllClustersWithOrgInfoByActiveDc() {
+        List<ClusterTbl> clusterTblList = clusterDao.findClustersWithOrgInfoByActiveDcId(1L);
+        clusterTblList.forEach(cluster->logger.info("{}", cluster));
+    }
+
+    @Override
+    protected String prepareDatas() throws IOException {
+        return prepareDatasFromFile("src/test/resources/cluster-dao-test.sql");
```
Comment (wenchaomeng) [inline redis/redis-console/src/test/java/com/ctrip/xpipe/redis/console/dao/ClusterDaoTest.java]: put it in code

File: redis/redis-console/src/test/java/com/ctrip/xpipe/redis/console/schedule/ScheduledOrganizationServiceTest.java [hunk 8]
```diff
@@ -0,0 +1,32 @@
...
+ * Created by zhuchen on 2017/8/31.
+ */
+@EnableScheduling
+public class ScheduledOrganizationServiceTest extends AbstractConsoleIntegrationTest {
+
+    @Autowired
+    ScheduledOrganizationService scheduledOrganizationService;
+
+    private void start() throws IOException, SQLException {
+        SpringApplication.run(ScheduledOrganizationServiceTest.class);
```
Comment (wenchaomeng) [inline redis/redis-console/src/test/java/com/ctrip/xpipe/redis/console/schedule/ScheduledOrganizationServiceTest.java]: spring already run

File: redis/redis-console/src/test/java/com/ctrip/xpipe/redis/console/service/impl/KeepercontainerServiceImplTest.java [hunk 9]
```diff
@@ -90,6 +90,6 @@ public  void testFindAllActiveByDcName(){
 
     @Override
     protected String prepareDatas() throws IOException {
-        return prepareDatasFromFile("src/test/resources/apptest.sql");
+        return prepareDatasFromFile("src/test/resources/keeper-container-service-impl-test.sql");
```
Comment (wenchaomeng) [inline redis/redis-console/src/test/java/com/ctrip/xpipe/redis/console/service/impl/KeepercontainerServiceImplTest.java]: put it in code?

File: redis/redis-console/src/test/java/com/ctrip/xpipe/redis/console/service/impl/OrganizationServiceImplTest.java [hunk 10]
```diff
@@ -0,0 +1,88 @@
...
+        List<OrganizationTbl> remoteDBOrgs = createOrganizationTblList(10)
+            .stream().map(org->org.setOrgName(org.getOrgName() + "-suffix")).collect(Collectors.toList());
+        List<OrganizationTbl> localDBOrgs = createOrganizationTblList(10);
+        List<OrganizationTbl> result = organizationService.getOrgTblUpdateList(remoteDBOrgs, localDBOrgs);
+        result.forEach(org->logger.info("org id: {}, org name: {}", org.getOrgId(), org.getOrgName()));
+        Assert.assertEquals(10, result.size());
+    }
+    @BeforeClass
+    public static void beforeOrganizationServiceImplTest() {
+        System.setProperty(CatConfig.CAT_ENABLED_KEY, "true");
```
Comment (wenchaomeng) [inline redis/redis-console/src/test/java/com/ctrip/xpipe/redis/console/service/impl/OrganizationServiceImplTest.java]: delete it?

Comment (NickNYU) [reply redis/redis-console/src/test/java/com/ctrip/xpipe/redis/console/service/impl/OrganizationServiceImplTest.java]: resolved

File: redis/redis-console/src/main/java/com/ctrip/xpipe/redis/console/controller/api/data/MetaUpdate.java (hunk 5 above)
Comment (NickNYU) [reply redis/redis-console/src/main/java/com/ctrip/xpipe/redis/console/controller/api/data/MetaUpdate.java]: resolved

File: redis/redis-console/src/main/java/com/ctrip/xpipe/redis/console/controller/api/data/MetaUpdate.java (hunk 4 above)
Comment (NickNYU) [reply redis/redis-console/src/main/java/com/ctrip/xpipe/redis/console/controller/api/data/MetaUpdate.java]: resolved

File: redis/redis-console/src/main/java/com/ctrip/xpipe/redis/console/config/impl/DefaultConsoleConfig.java (hunk 3 above)
Comment (NickNYU) [reply redis/redis-console/src/main/java/com/ctrip/xpipe/redis/console/config/impl/DefaultConsoleConfig.java]: resolved

File: redis/redis-console/src/main/java/com/ctrip/xpipe/redis/console/config/ConsoleConfig.java (hunk 2 above)
Comment (NickNYU) [reply redis/redis-console/src/main/java/com/ctrip/xpipe/redis/console/config/ConsoleConfig.java]: resolved

File: core/src/main/java/com/ctrip/xpipe/api/organization/Organization.java (hunk 1 above)
Comment (NickNYU) [reply core/src/main/java/com/ctrip/xpipe/api/organization/Organization.java]: resovled with all files

File: redis/redis-console/src/test/java/com/ctrip/xpipe/redis/console/service/impl/KeepercontainerServiceImplTest.java (hunk 9 above)
Comment (NickNYU) [reply redis/redis-console/src/test/java/com/ctrip/xpipe/redis/console/service/impl/KeepercontainerServiceImplTest.java]: resolved

File: redis/redis-console/src/test/java/com/ctrip/xpipe/redis/console/schedule/ScheduledOrganizationServiceTest.java (hunk 8 above)
Comment (NickNYU) [reply redis/redis-console/src/test/java/com/ctrip/xpipe/redis/console/schedule/ScheduledOrganizationServiceTest.java]: resovled

File: redis/redis-console/src/test/java/com/ctrip/xpipe/redis/console/dao/ClusterDaoTest.java (hunk 7 above)
Comment (NickNYU) [reply redis/redis-console/src/test/java/com/ctrip/xpipe/redis/console/dao/ClusterDaoTest.java]: resolved

**Pull request merged**